### PR TITLE
Save picture from history to the gallery

### PR DESCRIPTION
**Pull request by leonardvaney**
A popup menu with a single item appears when we click on an image of the history. We can then save this picture to the gallery by clicking on the single item from the menu.

### CODE REVIEW

**Comment by rkuenzi-epfl** [comment]
Do you think there is a (easy) way to attach the menu to the inside of the picture view? (I have a personal preference for the upper right corner)

File: app/src/main/java/com/github/wnder/HistoryAdapter.java [hunk 1]
```diff
@@ -80,6 +90,34 @@ public int getItemCount() {
...
+    private void saveToGallery(String pictureID, Bitmap bmp, ViewHolder holder){
+        if(pictureID.equals(Picture.UNINITIALIZED_ID)){
+            //Snack bar
+            Snackbar snackbar = Snackbar.make(holder.itemView, R.string.bar_save_is_impossible, BaseTransientBottomBar.LENGTH_SHORT);
+            snackbar.show();
+        }
+        else{
+            MediaStore.Images.Media.insertImage(holder.itemView.getContext().getContentResolver(), bmp, pictureID, "");
+            Snackbar snackbar = Snackbar.make(holder.itemView, R.string.bar_save_is_ok, BaseTransientBottomBar.LENGTH_SHORT);
+            snackbar.show();
```
Comment (rkuenzi-epfl) [inline app/src/main/java/com/github/wnder/HistoryAdapter.java:117]: You can display snackbars without a variable: 
```
Snackbar.make(holder.itemView, R.string.bar_save_is_ok, Snackbar.LENGTH_SHORT)
    .show();
```

**Comment by rkuenzi-epfl** [comment]
> I don't really know if it can appears in the image, mine is displayed just under the view it is associated with. I will see if it is possible.

We tried it together for some time and didn't find a way move it inside so it's fine like this.

File: app/src/main/java/com/github/wnder/HistoryAdapter.java (hunk 1 above)
Comment (leonardvaney) [reply app/src/main/java/com/github/wnder/HistoryAdapter.java]: It's done.

**Review by rkuenzi-epfl — APPROVED** [review]
I finally found out how to place the popup menu on top right corner of the image!
Please check it out and if you prefer your result just revert the changes.

File: app/src/androidTest/java/com/github/wnder/HistoryFragmentTest.java [hunk 2]
```diff
@@ -95,12 +86,28 @@ public void historyScoreboardIsClickable(){
...
 
         Intents.intended(hasComponent(ScoreboardActivity.class.getName()));
 
     }
+
+    @Test
+    public void historyPictureCanBeSaved(){
+        onView(withId(R.id.history_page)).perform(click());
+        onView(withId(R.id.historyImage)).perform(click());
+        onView(withText("Save to gallery")).perform(click());
```
Comment (JeremyDezalos) [inline app/src/androidTest/java/com/github/wnder/HistoryFragmentTest.java:111]: Nothing is tested here.

**Review by JeremyDezalos — CHANGES_REQUESTED** [review]
One test doesn't assert anything. Try to make it check if "save to gallery" is displayed

**Review by JeremyDezalos — APPROVED** [review]

**Pull request merged**